core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -148,8 +148,8 @@ public void createDirectory(CreateDirectoryPRequest request,
     CreateDirectoryPOptions options = request.getOptions();
     RpcUtils.call(LOG, () -> {
       AlluxioURI pathUri = getAlluxioURI(request.getPath());
-      mFileSystemMaster.createDirectory(pathUri,
-          CreateDirectoryContext.create(options.toBuilder()));
+      mFileSystemMaster.createDirectory(pathUri, (CreateDirectoryContext) CreateDirectoryContext
+          .create(options.toBuilder()).withTracker(new GrpcCallTracker(responseObserver)));
```

Comment:
if `withTracker` returns the generic type of the `OperationContext` (`C` in your implementation), then is it necessary to include the cast to `CreateDirectoryContext`? It just doesn't seem clean.


edit - I looked into it and figured out the issue. Poor use of generics. Change the class definition of `CreatePathContext` to remove the need to cast

old:

```java
public abstract class CreatePathContext<T extends GeneratedMessageV3.Builder<?>, K>
    extends OperationContext<T, CreatePathContext> {
```

new:
```java
public abstract class CreatePathContext<T extends GeneratedMessageV3.Builder<?>,
    K extends OperationContext> extends OperationContext<T, K> {
```